myhoard/backup_stream.py

```diff
@@ -675,10 +683,45 @@ def upload_progress(bytes_sent):
                 stats=self.stats,
             )
 
-        file_storage.store_file_object(
-            self._build_full_name("basebackup.xbstream"), stream, metadata=metadata, upload_progress_fn=upload_progress
-        )
+        split_nr = 0
+        uploaded_size = 0
+        basebackup_file_path = self._build_full_name("basebackup.xbstream")
+        while (split_nr == 0) or uploaded_size == self.split_size:
```

Comment:
If the last piece is exactly `self.split_size`, would it not cause any issues? It looks like it will still make an iteration over empty stream.

Reply:
We'll just upload an empty split-file which is fine. We could delete it but not sure we would want to add that complexity.
We'll also exit cleanly in case we're not splitting and the basebackup would be 0 bytes (for whatever reason that could happen), because of the `break` in the end. If that wouldn't be there we would have an issue.

Reply:
Just wanted to make sure this case is covered and tested.